### PR TITLE
Implemented naturalPrev function in Hierarchy class

### PR DESCRIPTION
**Pull request by jedateach**
This has been marked as "todo" since forever. I just need it for a project.

`naturalNext` function doesn't have unit tests, so they both need unit tests at some stage.

Only tested with parent->children relationship so far, rather than deep nesting.

### CODE REVIEW

File: model/Hierarchy.php [hunk 1]
```diff
@@ -701,12 +701,57 @@ public function getBreadcrumbs($separator = ' &raquo; ') {
...
+				}
+			}
+			
+			if($prevNode) {
+				return $prevNode;
+			}
+		}
+		
+		// if this is not an instance of the root class or has the root id, search the parent
+		if(!(is_numeric($root) && $root == $this->owner->ID || $root == $this->owner->class)
```
Comment (dhensby) [inline model/Hierarchy.php]: You need some more brackets here to group the conditions.

**Comment by dhensby** [comment]
I think this needs to go against 3 branch

File: model/Hierarchy.php (hunk 1 above)
Comment (tractorcow) [inline model/Hierarchy.php]: Is this enough?

```php
// if this is not an instance of the root class or has the root id, search the parent
if($root != $this->owner->ID && $root != $this->owner->class && ($parent = $this->owner->Parent())) {
	// ....
}
```

Comment (willmorgan) [inline model/Hierarchy.php]: `&& $parent->exists()`

**Comment by dhensby** [comment]
This also needs tests

**Comment by dhensby** [comment]
This actually needs to go against master as it's a breaking API change... even though it was never implemented :(

File: model/Hierarchy.php [hunk 2]
```diff
@@ -701,12 +701,57 @@ public function getBreadcrumbs($separator = ' &raquo; ') {
...
+				&& ($parent = $this->owner->Parent())) {
+			
+			return $parent->naturalPrev( $className, $root, $this->owner );
```
Comment (dhensby) [inline model/Hierarchy.php]: spacing inside parenthesis shouldn't be there

**Comment by dhensby** [comment]
@tractorcow is there any value in someone picking this up? I don't even know why there's a function stub for this

**Comment by tractorcow** [comment]
We should just remove the stub in 4.x.

**Pull request closed without merging**